osf/models/collection.py

```diff
@@ -39,6 +45,72 @@ class Meta:
     program_area = models.CharField(blank=True, max_length=127)
     school_type = models.CharField(blank=True, max_length=127)
     study_design = models.CharField(blank=True, max_length=127)
+    machine_state = models.IntegerField(
+        choices=CollectionSubmissionStates.int_field_choices(),
+        default=CollectionSubmissionStates.IN_PROGRESS,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.state_machine = CollectionSubmissionMachine(
+            model=self,
+            active_state=self.state,
+            state_property_name='state'
+        )
+
+    @property
+    def state(self):
+        '''Property to translate between ApprovalState Enum and DB string.'''
+        return CollectionSubmissionStates(self.machine_state)
+
+    @property
+    def provider(self):
+        return self.collection.provider
+
+    @property
+    def moderators(self):
```

Comment:
This is maybe not necessary. If it turns out to be helpful, can you union with the `admin` set?